pallets/gear/src/lib.rs

```diff
@@ -562,39 +569,30 @@ pub mod pallet {
                     )
                 };
 
-                core_processor::handle_journal(journal.clone(), &mut ext_manager);
+                // TODO: Check whether we charge gas fee for submitting code after #646
+                for note in journal.into_iter() {
```

Comment:
No need to explicitly call `.into_iter()`.
```suggestion
                for note in journal {
```